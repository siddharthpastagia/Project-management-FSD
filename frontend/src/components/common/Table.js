import "react-bootstrap-table-next/dist/react-bootstrap-table2.min.css";
import React from "react";
import BootstrapTable from "react-bootstrap-table-next";
import ToolkitProvider, { Search } from "react-bootstrap-table2-toolkit";
import "react-bootstrap-table2-toolkit/dist/react-bootstrap-table2-toolkit.min.css";

const Table = props => {
  const { SearchBar } = Search;
  const { data, columns, keyField, search, rowClickHandler } = props;
  const rowStyle = rowClickHandler
    ? { cursor: "pointer" }
    : { cursor: "default" };

  const rowClickEvents = {
    onClick: (e, row, rowIndex) => {
      if (rowClickHandler) {
        rowClickHandler(row, rowIndex);
      }
    }
  };

  return (
    <ToolkitProvider keyField="id" data={data} columns={columns} search>
      {props => (
        <div>
          <SearchBar {...props.searchProps} />
          <hr />
          <BootstrapTable
            striped
            condensed
            rowStyle={rowStyle}
            rowEvents={rowClickEvents}
            noDataIndication="No records found!!"
            {...props.baseProps}
          />
        </div>
      )}
    </ToolkitProvider>
  );
};

export default Table;
