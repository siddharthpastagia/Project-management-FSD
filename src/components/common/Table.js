import "react-bootstrap-table-next/dist/react-bootstrap-table2.min.css";
import React from "react";
import BootstrapTable from "react-bootstrap-table-next";
import ToolkitProvider, { Search } from "react-bootstrap-table2-toolkit";
import "react-bootstrap-table2-toolkit/dist/react-bootstrap-table2-toolkit.min.css";

const Table = props => {
  const { SearchBar } = Search;
  const { data, columns, keyField, search } = props;
  return (
    <ToolkitProvider keyField="id" data={data} columns={columns} search>
      {props => (
        <div>
          <SearchBar {...props.searchProps} />
          <hr />
          <BootstrapTable
            striped
            condensed
            noDataIndication="No records"
            {...props.baseProps}
          />
        </div>
      )}
    </ToolkitProvider>
  );
};

export default Table;
