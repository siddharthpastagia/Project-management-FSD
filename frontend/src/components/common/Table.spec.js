import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import Table from "./Table";

describe("Table Component", () => {
  afterEach(cleanup);
  const data = [{ projectName: "A" }, { projectName: "B" }];
  const columns = [{ dataField: "projectName", text: "Project List" }];
  const rowClickHandler = jest.fn();
  it("Should render Table component", () => {
    const { container } = render(<Table data={data} columns={columns} />);
    expect(container).toMatchSnapshot();
  });

  it("should click on row", () => {
    const { container, getByText } = render(
      <Table
        data={data}
        columns={columns}
        rowClickHandler={rowClickHandler(data[0].name)}
      />
    );
    const firstRow = getByText("A");
    fireEvent.click(firstRow);
    expect(container).toMatchSnapshot();
  });
});
