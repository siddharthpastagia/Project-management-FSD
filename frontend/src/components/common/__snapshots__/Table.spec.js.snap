// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Should render Table component 1`] = `
<div>
  <div>
    <label
      class="search-label"
      for="search-bar-0"
    >
      <span
        class="sr-only"
      >
        Search this table
      </span>
      <input
        aria-label="enter text you want to search"
        class="form-control "
        id="search-bar-0"
        placeholder="Search"
        type="text"
        value=""
      />
    </label>
    <hr />
    <div
      class="react-bootstrap-table"
    >
      <table
        class="table table-striped table-bordered table-condensed"
      >
        <thead>
          <tr>
            <th
              tabindex="0"
            >
              Project List
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            style="cursor: default;"
          >
            <td>
              A
            </td>
          </tr>
          <tr
            style="cursor: default;"
          >
            <td>
              B
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Table Component should click on row 1`] = `
<div>
  <div>
    <label
      class="search-label"
      for="search-bar-0"
    >
      <span
        class="sr-only"
      >
        Search this table
      </span>
      <input
        aria-label="enter text you want to search"
        class="form-control "
        id="search-bar-0"
        placeholder="Search"
        type="text"
        value=""
      />
    </label>
    <hr />
    <div
      class="react-bootstrap-table"
    >
      <table
        class="table table-striped table-bordered table-condensed"
      >
        <thead>
          <tr>
            <th
              tabindex="0"
            >
              Project List
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            style="cursor: default;"
          >
            <td>
              A
            </td>
          </tr>
          <tr
            style="cursor: default;"
          >
            <td>
              B
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
